### ISSUE TITLE
Middleware is not triggered with certain http method on controller's sub-routes

### ISSUE THREAD

**Issue by ianzone** [nestjs/nest#11850]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When accessing localhost:3000/users/rsvp the POST, PUT, OPTION, and HEAD methods won't trigger the middleware. 

### Minimum reproduction code

https://github.com/ianzone/nest-issue

### Steps to reproduce

1. pnpm i
2. pnpm run dev
3. access localhost:3000/users/rsvp by POST, PUT, OPTION, and HEAD methods using postman

### Expected behavior

terminal logs out "AuthMiddleware invoked..."

### Package

- [ ] I don't know. Or some 3rd-party package
- [X] <code>@nestjs/common</code>
- [X] <code>@nestjs/core</code>
- [ ] <code>@nestjs/microservices</code>
- [ ] <code>@nestjs/platform-express</code>
- [X] <code>@nestjs/platform-fastify</code>
- [ ] <code>@nestjs/platform-socket.io</code>
- [ ] <code>@nestjs/platform-ws</code>
- [ ] <code>@nestjs/testing</code>
- [ ] <code>@nestjs/websockets</code>
- [ ] Other (see below)

### Packages versions

```
[System Information]
OS Version     : Linux 5.15
NodeJS Version : v18.16.0
PNPM Version    : 8.6.3

[Nest CLI]
Nest CLI Version : 10.0.3

[Nest Platform Information]
platform-fastify version : 10.0.2
mapped-types version     : 2.0.0
schematics version       : 10.0.1
testing version          : 10.0.2
common version           : 10.0.2
core version             : 10.0.2
cli version              : 10.0.3
```

### Node.js version

v18.16.0

### In which operating systems have you tested?

- [ ] macOS
- [ ] Windows
- [X] Linux

### Other

https://github.com/nestjs/nest/issues/11833

**Comment by Artuomka** [comment]
"The same problem occurs with platform-express. It arises after updating the package version of @nestjs/core to 10."

**Comment by kamilmysliwiec** [comment]
Fixed in 10.0.3. Thanks for reporting!
